### ISSUE TITLE
The Picker has gone nuts in V5.6 Beta!

### ISSUE THREAD

**Issue by iamjaydear** [RandomEngy/VidCoder#524]
In the Picker window, when I select a preset, do nothing to it and then move to another I'm asked if I want to save changes, but I made none, so I say no and then the same thing happens on the next one, etc. At some point in this sequence VC locks up to any further user control and the title bar of the Encoding Settings window flickers at amazingly high speed between the last two Picker selections I made. The only way out is to kill VC in Process Explorer or Task Manager.

**Comment by iamjaydear** [comment]
I think I see where I went wrong... I forgot you can edit but not save the Default preset. If you edit it and then do not save the "Picker x" that appears at the bottom of your list, and then select another preset, it goes nuts as described above. VidCoder abhors a vacuum :P

**Comment by iamjaydear** [comment]
If you have a long list of presets, a new one generated from the Default one goes to the bottom of the list. Maybe it'd be good if it appeared at the top, say as "- Picker xx", with the dash (-) causing it to sort to the top.

**Comment by RandomEngy** [comment]
If you have a series of steps to reproduce reliably, let me know. I've tried doing what you describe but I don't get that flickering. Right now there is no way to change the sort order of the Pickers. Initially I didn't think people would have so many that sorting or other management would be needed. How many do you have?

**Comment by iamjaydear** [comment]
I think I'm suffering withdrawal symptoms due to the "More Settings" changes :P There's also an identity problem between Picker Presets and Encoding Presets in my head.  If I have a picker in use and turn an item in it's encoding setting off and back on I feel I've cancelled the change, but VC disagrees unlike some other s/w I use.  So a day or two later when I need a different picker VC asks if I want to save my preset - is that a picker preset or an encoder preset (that was yesterday, how do I know?). So I'm left wondering what I changed (nothing in fact), hit No to be safe and then when I select another picker I sometimes get the 'logic loop' flickering.

I can have a few dozen pickers at times. That comes about when I have to make another special one to do some custom cropping / padding / resizing / etc. for just a few videos. I try to avoid having any "you made a change asterisks" by doing this, but as I say above a little test change even though its undone right away does not cancel the asterisk. I don't like saving a setting when I'm not sure any more what VC thinks I did.

These special pickers get cleaned out occasionally. Sometimes I get a surprise result and keep that one with a new name which is where the desire to re-sort the list comes in :- trying to get things logically grouped. Just being able to slide them up and down in the list like the encoding list would be really good.

**Comment by RandomEngy** [comment]
Hmm. The cropping/sizing settings are on the encoder presets, which you can put into a tree structure. Hit ... -> Create subfolder. That can help you keep them more organized.

As for reordering the pickers, I do think that would be a good idea. I can look at adding that.

**Comment by iamjaydear** [comment]
It took me a while to find 'Create subfolder' - it's only available in the main window, not in the side bar in the encoding settings window - I'll give it a try soon. Thing is, if I change the encoding preset in a picker the picker name needs to change to reflect the difference, otherwise I'll go and do a bunch of encodes that have to be redone because I forgot the little change/s I made.  

The flickering problem occurred when I changed the Default Picker for a quickie encode and then tried to cancel because I didn't want to save it. I think the cancel command was the trigger for the flickering. There was a bit of confusion because the changed one appears at the bottom of the list and was not visible. If I change one of mine I'm more mindful and save it as a new one. I thought I could temporarily change the default one without saving, but not possible. I doubt it's a real problem. I'll try to reproduce, but I'll need to record the screen because my mental steps get lost in the heat of the moment.

Picker slider will be good :)
